### ISSUE TITLE
Bug in checkConfiguration leads to EDT assertion

### ISSUE THREAD

**Issue by abrooksv** [aws/aws-toolkit-jetbrains#979]
```
Access is allowed from event dispatch thread only.

com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@a5adf7a
Current thread: Thread[OurIconsCalculating Pool,4,Idea Thread Group] 1222850060
SystemEventQueueThread: Thread[AWT-EventQueue-0 2019.1#IC-191.6183.87 IDEA, eap:false, os:Mac OS X 10.12.6, java-version:JetBrains s.r.o 1.8.0_202-release-1483-b39,6,Idea Thread Group] 841377470
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1128)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1117)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveAllDocuments(FileDocumentManagerImpl.java:288)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveAllDocuments(FileDocumentManagerImpl.java:281)
	at software.aws.toolkits.jetbrains.services.lambda.execution.LambdaRunConfigurationBase.resolveInput(LambdaRunConfigurationBase.kt:51)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.LocalLambdaRunConfiguration.checkConfiguration(LocalLamdaRunConfiguration.kt:73)
	at com.intellij.execution.impl.RunnerAndConfigurationSettingsImpl.checkSettings(RunnerAndConfigurationSettingsImpl.kt:310)
	at com.intellij.execution.RunnerAndConfigurationSettings.checkSettings(RunnerAndConfigurationSettings.java:116)
	at com.intellij.execution.impl.TimedIconCache.calcIcon(TimedIconCache.kt:71)
```

**Comment by abrooksv** [comment]
`checkConfiguration` should not have to resolve the inputs, it should be good enough the file exists

**Comment by abrooksv** [comment]
Fixed in 1.4

**Comment by chkothe** [comment]
I'm on 1.4, but I'm getting pretty much the same stack trace every now and then.
This is with PyCharm 2019.1.3 Professional Edition on Win10 x64.

```
Access is allowed from event dispatch thread only.

com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@4693b6ee
Current thread: Thread[OurIconsCalculating Pool,4,Idea Thread Group] 757175145
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,Idea Thread Group] 109216679
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1128)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1117)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveAllDocuments(FileDocumentManagerImpl.java:288)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveAllDocuments(FileDocumentManagerImpl.java:281)
	at software.aws.toolkits.jetbrains.services.lambda.execution.LambdaRunConfigurationBase.resolveInput(LambdaRunConfigurationBase.kt:51)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.LocalLambdaRunConfiguration.checkConfiguration(LocalLamdaRunConfiguration.kt:73)
	at com.intellij.execution.impl.RunnerAndConfigurationSettingsImpl.checkSettings(RunnerAndConfigurationSettingsImpl.kt:310)
	at com.intellij.execution.RunnerAndConfigurationSettings.checkSettings(RunnerAndConfigurationSettings.java:116)
	at com.intellij.execution.impl.TimedIconCache.calcIcon(TimedIconCache.kt:71)
	at com.intellij.execution.impl.TimedIconCache.access$calcIcon(TimedIconCache.kt:18)
	at com.intellij.execution.impl.TimedIconCache$get$$inlined$write$lambda$1.fun(TimedIconCache.kt:50)
	at com.intellij.execution.impl.TimedIconCache$get$$inlined$write$lambda$1.fun(TimedIconCache.kt:18)
	at com.intellij.ui.DeferredIconImpl.evaluate(DeferredIconImpl.java:280)
	at com.intellij.ui.DeferredIconImpl.lambda$null$0(DeferredIconImpl.java:165)
	at com.intellij.ui.IconDeferrerImpl.evaluateDeferred(IconDeferrerImpl.java:111)
	at com.intellij.ui.DeferredIconImpl.lambda$null$1(DeferredIconImpl.java:165)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1166)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:68)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:121)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:86)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:110)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:68)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:89)
	at com.intellij.ui.DeferredIconImpl.lambda$paintIcon$4(DeferredIconImpl.java:164)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:220)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.lambda$run$0(BoundedTaskExecutor.java:198)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:224)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:194)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by abrooksv** [comment]
@chkothe you are correct, this was mis-labled. It is fixed in master and will be in 1.5
